packages/amplify-cli-core/src/feature-flags/featureFlagProvider.ts

```diff
@@ -0,0 +1,422 @@
+      projectPath,
+    });
+
+    // Load the flags from all providers
+    const fileFlags = await fileValueProvider.load();
+    const envFlags = instance.transformEnvFlags(await envValueProvider.load());
+
+    // Validate the loaded flags from all providers
+    instance.validateFlags([
+      {
+        name: 'File',
+        flags: fileFlags,
+      },
+      {
+        name: 'Environment',
+        flags: envFlags,
+      },
+    ]);
+
+    // Build default value objects from registrations
+    instance.buildDefaultValues();
+
+    //
+    // To make access easy, we are unfolding the values from the providers, dot notation based key value pairs.
+    // Example: 'graphqltransformer.transformerversion': 5
+    //
+    // top to bottom the following order is used:
+    // - file project level
+    // - file env level
+    // - environment project level
+    // - environment env level
+    //
+    // The sections, properties and values are verified against the dynamically built up yup schema
+    //
+
+    instance.effectiveFlags = _.merge(
+      {},
+      fileFlags.project,
+      fileFlags.environments[environmentProvider.getCurrent()] ?? {},
+      envFlags.project,
+      envFlags.environments[environmentProvider.getCurrent()] ?? {},
+    );
+
+    FeatureFlagProvider.instance = instance;
+
+    return instance;
+  };
+
+  /**
+   * This method reads the project configuration file if exist and adds the features section based on the default feature flags file.
+   * If the configuration file does not exist it will be created with the default features. If the configuration file exists and already has a
+   * features section it will be preserved and will not be overwritten.
+   *
+   * @param newProject True if settings for new projects requested or false if default for existing project are requested
+   */
+  public static ensureDefaultFeatureFlags = async (newProject: boolean): Promise<void> => {
+    FeatureFlagProvider.ensureInitialized();
+
+    const configFileName = path.join(FeatureFlagProvider.instance.projectPath, AmplifyConfigFileName);
+
+    let config = await JSONUtilities.readJson<{ [key: string]: any }>(configFileName, {
+      throwIfNotExist: false,
+      preserveComments: true,
+    });
+
+    // If no configuration file or configuration file exists but does not have a features property create it and overwrite the file
+    if (!config || (config && !config.features)) {
+      config = {
+        ...config,
+        features: newProject ? FeatureFlagProvider.getNewProjectDefaults() : FeatureFlagProvider.getExistingProjectDefaults(),
+      };
+
+      await JSONUtilities.writeJson(configFileName, config, {
+        keepComments: true,
+      });
+    }
+  };
+
+  private static ensureInitialized = (): void => {
+    if (!FeatureFlagProvider.instance) {
+      throw new Error('FeatureFlagsProvider is not initialized');
+    }
+  };
+
+  public static getBoolean = (flagName: string): boolean => {
```

Comment:
why make all of these methods static? If everything's going to be static then you don't need the class. But I would refactor this file to have a factory function that returns a singleton instance of the class and make all of these instance methods

To pull that off, the current env and project path would also have to be stateful imports, but I think that also makes sense, especially now that that could go in amplify core